### PR TITLE
Add `Text` class for `font_file` as a parameter to multiple language

### LINKED ISSUE #4548: User defined font and Chinese Characters supoort
### Describe the feature you would like to be added.

I found Pyvista cannot support Chinese Characters. I found a similar issue [#3501](https://github.com/pyvista/pyvista/issues/3501), but it didn't give a solution. It seems the solution should be Pyvista can support a font that has Chinese Characters such as "Microsoft Yahei". However, in the latest version 0.39, Pyvista can only set the font to "courier", "times" and "arial".
So I'm wondering if there is a simple to solve this problem or if a new feature should be added.

### Links to VTK Documentation, Examples, or Class Definitions.

I found something about [vtk-chinese-font](https://discourse.slicer.org/t/how-to-show-chinese-characters-in-vtk-widget/19305).

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

resolves #4548

```python
import os
import pyvista


plotter = pyvista.Plotter()
font_file = "fonts/Mplus2-Regular.ttf"
plotter.add_text("左上", position='upper_left', font_size=25, color='blue', font_file=font_file)
plotter.add_text("中央", position=(0.5, 0.5), viewport=True, orientation=-90, font_file=font_file)
plotter.show()
```

add_text_font_file

### Details

- [x] Corresponds to a local path.
- [x] Support for errors when a file does not exist

### CODE REVIEW

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by MatthewFlamm** [comment]
This is nice.  I'm foreseeing that people will want this to also work for text outside of `add_text` for example the scalar bar. Long term we may want to have a pyvista TextProperty class, for which we can set defaults in the theme.

**Comment by banesullivan** [comment]
I wonder if we should create a wrapping of the TextureActor class that has a hook to set these font files on construction? That way we might be able to register a font file with the theme so this will "just work" anywhere text is added? Just a thought...

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
The following two files are used as coding references.
- pyvista/plotting/actor.py
- pyvista/plotting/_property.py

File: tests/plotting/test_plotting.py [hunk 1]
```diff
@@ -2637,6 +2637,14 @@ def test_add_text_latex():
     plotter.show()
 
 
+def test_add_text_font_file():
+    plotter = pyvista.Plotter()
```
Comment (tkoyama010) [inline tests/plotting/test_plotting.py:2641]:
```suggestion
    plotter = pv.Plotter()
```

File: pyvista/plotting/text.py [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+        """
+        return Color(self.GetBackgroundColor())
+
+    @background_color.setter
+    def background_color(self, value):
+        """Return or set the background color of text's property.
+
+        Parameters
+        ----------
+        value : str | Color
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:413]:
```suggestion
        value : ColorLike
```

File: pyvista/plotting/text.py [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+            self.orientation = orientation
+        if font_size is not None:
+            self.font_size = font_size
+        if font_file is not None:
+            self.set_font_file(font_file)
+        if shadow:
+            self.enable_shadow()
+
+    @property
+    def color(self) -> ColorLike:
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:340]:
```suggestion
    def color(self) -> Color:
```

File: pyvista/plotting/text.py [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+        """Return the color of text's property.
+
+        Returns
+        -------
+        ColorLike
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:345]:
```suggestion
        Color
```

File: pyvista/plotting/text.py [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+            The color of text's property.
+
+        """
+        return Color(self.GetColor())
+
+    @color.setter
+    def color(self, color):
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:352]:
```suggestion
    def color(self, colo : ColorLiker):
```

File: pyvista/plotting/text.py [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+        value : float
+            Opacity of the text. A single float value that will be applied globally
+            opacity of the text and uniformly applied everywhere. Between 0 and 1.
+
+        """
+        _check_range(value, (0, 1), 'opacity')
+        self.SetOpacity(value)
+
+    @property
+    def background_color(self):
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:396]:
```suggestion
    def background_color(self) -> Color:
```

File: pyvista/plotting/text.py [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+        Returns
+        -------
+        Color
+            The background color of text's property.
+
+        """
+        return Color(self.GetBackgroundColor())
+
+    @background_color.setter
+    def background_color(self, value):
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:408]:
```suggestion
    def background_color(self, color: ColorLike):
```

File: pyvista/plotting/text.py [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
+        """Return or set the background color of text's property.
+
+        Parameters
+        ----------
+        value : ColorLike
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:413]:
```suggestion
        color : ColorLike
```

File: pyvista/plotting/text.py [hunk 9]
```diff
@@ -0,0 +1,611 @@
...
+            Either a string, RGB list, or hex color string.  For example:
+            ``color='white'``, ``color='w'``, ``color=[1.0, 1.0, 1.0]``, or
+            ``color='#FFFFFF'``. Color will be overridden if scalars are
+            specified.
+
+        """
+        self._background_color_set = value is not None
+        rgb_color = Color(value)
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:421]:
```suggestion
        rgb_color = Color(color)
```

File: pyvista/plotting/text.py [hunk 10]
```diff
@@ -0,0 +1,611 @@
...
+        Parameters
+        ----------
+        value : ColorLike
+            Either a string, RGB list, or hex color string.  For example:
+            ``color='white'``, ``color='w'``, ``color=[1.0, 1.0, 1.0]``, or
+            ``color='#FFFFFF'``. Color will be overridden if scalars are
+            specified.
+
+        """
+        self._background_color_set = value is not None
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:420]:
```suggestion
        self._background_color_set = color is not None
```

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
@pyvista/developers It is ready for review :)

**Review by akaszynski — APPROVED** [review]
Great work here @tkoyama010. I added an example within the `add_text` docstring and changed "free type" to "freetype", but other than that it looks great!

**Comment by akaszynski** [comment]
Let's merge this and all other approved PRs and get the next release out tomorrow.

**Comment by tkoyama010** [comment]
@akaszynski
Thanks a lot. I am really happy that we can add this to release.
Could you let me create release branch? I have never made it all the way through the pyvista release procedure and would like to challenge it.

**Comment by akaszynski** [comment]
> Could you let me create release branch?

Sure!

Best practice is to create one once all the approved PRs have been merged. We can shorten the 24H rule if you'd like to do it today. You can then use the [Automatically generated release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes) to help you draft the release notes. I typically use the release branch to first bump the next dev release number by merging to main and then (after undeleting it), bump to the stable next release number and then tag it.

**Comment by tkoyama010** [comment]
Thanks! I will try it tomorrow.

**Pull request merged**